File: typescript/src/utils/banner.ts
/**
 * ASCII banner utilities for Conclave CLI.
 */

import figlet from 'figlet';
import chalk from 'chalk';

// Gradient colors from yellow to orange
const GRADIENT_COLORS = [
    chalk.hex('#FFFF00'),  // bright yellow
    chalk.hex('#FFD700'),  // gold/yellow
    chalk.hex('#FFA500'),  // orange
    chalk.hex('#FF8C00'),  // dark orange
    chalk.hex('#FF4500'),  // orange red
];

/**
 * Print the Conclave ASCII art banner with gradient colors.
 */
export function printBanner(subtitle?: string): void {
    const asciiArt = figlet.textSync('CONCLAVE', { font: 'Banner3' });
    const lines = asciiArt.split('\n');

    // Print with left-to-right gradient effect
    lines.forEach((line) => {
        if (line.trim()) {
            const trimmedLine = line.trimEnd();
            const lineLen = trimmedLine.length;
            let coloredLine = '';
            for (let j = 0; j < trimmedLine.length; j++) {
                // Calculate color index based on horizontal position
                const colorIdx = Math.floor(j / Math.max(lineLen - 1, 1) * (GRADIENT_COLORS.length - 1));
                const color = GRADIENT_COLORS[colorIdx];
                coloredLine += color.bold(trimmedLine[j]);
            }
            console.log(coloredLine);
        }
    });

    // Print subtitle
    if (subtitle) {
        console.log(chalk.gray(subtitle));
    } else {
        console.log(chalk.gray('Multi-LLM collaboration CLI'));
    }
}

/**
 * Get the raw ASCII banner text without colors.
 */
export function getBannerText(): string {
    return figlet.textSync('CONCLAVE', { font: 'Banner3' });
}

File: python/janus/utils/banner.py
"""ASCII banner utilities for Janus CLI."""

import pyfiglet
from rich.console import Console


# Gradient colors from yellow to orange
GRADIENT_COLORS = [
    "bright_yellow",
    "yellow",
    "orange1",
    "dark_orange",
    "orange_red1",
]


def print_banner(console: Console | None = None, subtitle: str | None = None) -> None:
    """Print the Janus ASCII art banner with gradient colors.

    Args:
        console: Rich console to print to. Creates new one if not provided.
        subtitle: Optional subtitle to display under the banner.
    """
    if console is None:
        console = Console()

    # Generate ASCII art (banner3 = the # style)
    ascii_art = pyfiglet.figlet_format("JANUS", font="banner3")
    lines = ascii_art.split("\n")

    # Print with gradient effect
    for i, line in enumerate(lines):
        if line.strip():
            color = GRADIENT_COLORS[i % len(GRADIENT_COLORS)]
            console.print(line, style=f"bold {color}")

    # Print subtitle
    if subtitle:
        console.print(f"[dim]{subtitle}[/dim]")
    else:
        console.print("[dim]Multi-LLM collaboration CLI[/dim]")


def get_banner_text() -> str:
    """Get the raw ASCII banner text without colors.

    Returns:
        The ASCII art banner as a string.
    """
    return pyfiglet.figlet_format("JANUS", font="banner3")
